### PR TITLE
Bug 1559411 Reconfigure per-namespace republishing

### PR DESCRIPTION
**Pull request by jklukas**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1559411

### CODE REVIEW

File: ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublishPerNamespace.java [hunk 1]
```diff
@@ -37,13 +37,7 @@ public PDone expand(PCollection<PubsubMessage> input) {
     for (int i = 0; i < numDestinations; i++) {
       Destination destination = destinations.get(i);
       RepublisherOptions.Parsed opts = baseOptions.as(RepublisherOptions.Parsed.class);
-
-      // The destination pattern here must be compile-time due to a detail of Dataflow's
-      // streaming PubSub producer implementation; if that restriction is lifted in the future,
-      // this can become a runtime parameter and we can perform replacement via NestedValueProvider.
```
Comment (relud) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublishPerNamespace.java]: I think we should preserve this comment

File: ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublisherOptions.java [hunk 2]
```diff
@@ -93,19 +93,13 @@
...
   void setPerDocTypeDestination(String value);
 
-  @Description("A comma-separated list of namespaces that should be republished to individual"
-      + " topics")
-  List<String> getPerNamespaceEnabledList();
+  @Description("A JSON-formatted map of document namespaces to output topic names (assuming"
+      + " --outputType=pubsub) for per-namespace sampling; the verbose map representation is used"
+      + " here to support cases where the destination topics are heterogeneous and may live in"
+      + " different projects")
+  Map<String, String> getPerNamespaceDestinations();
```
Comment (relud) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublisherOptions.java]: I don't think it's worth blocking this PR, but to reduce the amount of repetition here maybe we could accept `{"<destination_pattern>":["<namespace>"]}` e.g.
```json
{
  "project/telemetry/topic/per-namespace-${document_namespace}":[
    "telemetry",
    "glean"
  ],
  "project/pocket/topic/namespace-pocket":[
     "pocket"
  ]
}
```

File: ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublishPerNamespace.java (hunk 1 above)
Comment (jklukas) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublishPerNamespace.java]: This comment still exists in the per-doctype transform.

File: ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublisherOptions.java (hunk 2 above)
Comment (jklukas) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublisherOptions.java]: Hmmm... Interesting. I will think on this.

Comment (jklukas) [reply ingestion-beam/src/main/java/com/mozilla/telemetry/republisher/RepublisherOptions.java]: Broke this out to https://github.com/mozilla/gcp-ingestion/issues/677

**Pull request merged**